plugins/modules/host.py

```diff
@@ -256,13 +453,41 @@ def main():
         elif 'owner_group' in module.foreman_params:
             module.foreman_params['owner_type'] = 'Usergroup'
 
+        if 'interfaces_attributes' in module.foreman_params:
+            filtered = [nic for nic in ({k: v for k, v in obj.items() if v} for obj in module.foreman_params['interfaces_attributes']) if nic]
+            module.foreman_params['interfaces_attributes'] = filtered
+
     with module.api_connection():
+        entity = module.lookup_entity('entity')
+
         if not module.desired_absent:
             module.auto_lookup_entities()
+
+            if 'interfaces_attributes' in module.foreman_params:
+                interfaces = []
+                for interface in module.foreman_params['interfaces_attributes']:
+                    for key, entity_spec in interfaces_foreman_spec.items():
+                        if key in interface and entity_spec.get('resolve', True) and entity_spec.get('type') in {'entity', 'entity_list'}:
+                            interface[key] = module._lookup_entity(interface[key], entity_spec)
+                    if not entity:
+                        interface = _flatten_entity(interface, interfaces_foreman_spec)
+                    interfaces.append(interface)
+                module.foreman_params['interfaces_attributes'] = interfaces
+
+        if entity and 'interfaces_attributes' in module.foreman_params:
```

Comment:
this logic is determining whether we add the interfaces at create or manipulate them after update?

I think this is worth a comment.